packages/react-fresh/src/__tests__/ReactFreshBabelPlugin-test.js

```diff
@@ -186,4 +186,103 @@ describe('ReactFreshBabelPlugin', () => {
     `),
     ).toMatchSnapshot();
   });
+
+  it('registers likely HOCs with inline functions', () => {
+    expect(
+      transform(`
+        const A = forwardRef(function() {
+          return <h1>Foo</h1>;
+        });
+        const B = memo(React.forwardRef(() => {
+          return <h1>Foo</h1>;
+        }));
+        export default React.memo(forwardRef((props, ref) => {
```

Comment:
I was initially surprised that the anonymous inner functions in `A` and `B` get registered, but the default export's anonymous function doesn't. I guess this is because the first two cases have a name that we can track back to and the third one doesn't.

_But_ if we changed the third one to be a named function, it would still get skipped.

This would match:
```js
export default function Foo() {
  return null;
}
```
And this would match:
```js
const Foo = function Foo(props, ref) {
  return null;
}
export default forwardRef(Foo);
```
But this would not:
```js
export default forwardRef(function Foo(props, ref) {
  return null;
});
```

I think the latter is a pattern I've used from time to time! 😁

Reply:
Fixed